### DISCUSSION TITLE
How to delete a single tag from a remote repository, if there are multiple ones pointing to the same manifest?

### DISCUSSION THREAD

**Discussion by Exagone313** [oras-project/oras#1594] [Q&A]
Hello,

When you pass a tag to `oras manifest delete` and that multiple tags point to the same manifest, this command will delete all the tags associated with this manifest.

Is there a way to only delete the tag if there are multiple tags pointing to the same manifest?

For building a workaround, is there a way to delete a single tag (without the associated manifest)? Apparently it can be done by sending a `DELETE` HTTP request to `${REGISTRY}/v2/${REPOSITORY}/manifests/${TAG}`. That way, a **non-atomic** way to do this job would be to list the tags and run a different delete command if there are more than one tags. (I think there isn't an atomic way to do this with the registry API, unless you rely on garbage collecting untagged manifests.)

I have looked at the source code and I don't think it exists, could be transformed into a feature suggestion issue.

**Comment by FeynmanZhou** [comment]
Hi @Exagone313 ,

Apologies for the late response. Could you please elaborate more on the case "multiple tags pointing to the same manifest"? Can you give an example? I am trying to figure out you scenario and identify whether this should be a feature request. Thanks

**Reply by Exagone313** [reply]
@FeynmanZhou Here is a quick example:

```bash
podman build -t registry.example/dummy:foo .
podman tag registry.example/dummy:foo  registry.example/dummy:bar
podman push registry.example/dummy:foo
podman push registry.example/dummy:bar
oras SOME_COMMAND delete registry.example/dummy:foo
```

### expected result

* `registry.example/dummy:foo` is deleted
* `registry.example/dummy:bar` is **not** deleted

### actual result with `oras manifest delete`

* `registry.example/dummy:foo` is deleted
* `registry.example/dummy:bar` is deleted

---

My use case here is to push a container image (or other types of manifest) with a version number and an environment name as tags pointing to this same image. If I want to create an ephemeral environment with its own tags and then clean it up, it would end up deleting the tags with version numbers unexpectedly.

**Comment by rcarre** [comment]
Dear @Exagone313 & @FeynmanZhou,
would you any update on this topic ? Thx

I am facing a similar issue with 2 artifact tags  with same digest @sha256:

- `7.26.10-20260110070007-sbom`
- `sha256:bd26540869d516c79f20e688a9eaff03b4cbfd296aa899f19c03f97a7d98b743`

```
oras manifest delete myArtifact@sha256:bd26540869d516c79f20e688a9eaff03b4cbfd296aa899f19c03f97a7d98b743

Are you sure you want to delete the manifest "sha256:330078eb7ee187e1dedf9bc237deb37cdb197cec7135c6a24a5b0c995db1fb45" and all tags associated with it? [y/N] y
Deleted [registry] myArtifact@sha256:330078eb7ee187e1dedf9bc237deb37cdb197cec7135c6a24a5b0c995db1fb45
```

Then, both tags are deleted ... I expected to keep the tag `7.26.10-20260110070007-sbom` albeit the alert **and all tags associated with it** in the above confirmation message ...

Here is my `oras version` output:
```
Version:        1.3.2
Go version:     go1.26.2
OS/Arch:        linux/amd64
Git commit:     fe425992fdfdf300a1cfb380bc4271b3e1a3d3db
Git tree state: clean
```
